typescript/packages/subsurface-viewer/src/layers/map/mapLayer.ts

```diff
@@ -43,14 +43,70 @@ type Frame = {
 };
 
 export type Params = [
-    meshData: Float32Array,
-    propertiesData: Float32Array,
+    meshData: Float32Array | null,
+    propertiesData: Float32Array | null,
     isMesh: boolean,
     frame: Frame,
     smoothShading: boolean,
     gridLines: boolean,
 ];
 
+async function loadURLData(url: string): Promise<Float32Array | null> {
```

Comment:
Shouldn't these function be shared and used in every place where data arrays are loaded ?

Reply:
Yes, they should be but it's more preferable to refactor in a dedicated PR rather than in this bug fixing one.